src/test/java/org/opentripplanner/middleware/models/MonitoredTripTest.java

```diff
@@ -41,4 +47,81 @@ void initializeFromItineraryAndQueryParamsShouldNotModifyModes() {
         trip.initializeFromItineraryAndQueryParams(variables);
         assertEquals(originalModes, trip.otp2QueryParams.modes);
     }
+
+    @ParameterizedTest
+    @MethodSource("createGetAddedUsersCases")
+    void canGetAddedUsers(MonitoredTrip.TripUsers originalUsers, MonitoredTrip.TripUsers finalUsers, MonitoredTrip.TripUsers expected) {
+        MonitoredTrip originalTrip = new MonitoredTrip();
+        originalTrip.primary = originalUsers.primary;
+        originalTrip.companion = originalUsers.companion;
+        originalTrip.observers = originalUsers.observers;
+
+        MonitoredTrip finalTrip = new MonitoredTrip();
+        finalTrip.primary = finalUsers.primary;
+        finalTrip.companion = finalUsers.companion;
+        finalTrip.observers = finalUsers.observers;
+
+        MonitoredTrip.TripUsers tripUsers = MonitoredTrip.getAddedUsers(finalTrip, originalTrip);
+
+        assertEquals(expected.primary != null, tripUsers.primary != null);
+        if (expected.primary != null && tripUsers.primary != null) {
+            assertEquals(expected.primary.userId, tripUsers.primary.userId);
+        }
+        assertEquals(expected.companion != null, tripUsers.companion != null);
+        if (expected.companion != null && tripUsers.companion != null) {
+            assertEquals(expected.companion.email, tripUsers.companion.email);
+        }
+        assertEquals(expected.observers, tripUsers.observers);
+    }
+
+    private static Stream<Arguments> createGetAddedUsersCases() {
+        MobilityProfileLite primary = new MobilityProfileLite();
+        primary.userId = "primary-user-id";
+
+        MobilityProfileLite newPrimary = new MobilityProfileLite();
+        primary.userId = "new-primary-user-id";
```

Comment:
`newPrimary ` and not `primary`

Reply:
Good catch, thank you! (https://github.com/ibi-group/otp-middleware/pull/273/commits/02d74f9db32244b6de46ab1d8be8d9aad86b8708)